src/Raven.Client/Documents/Session/ClusterTransactionSession.cs

```diff
@@ -8,105 +9,87 @@ namespace Raven.Client.Documents.Session
 {
     public abstract class ClusterSessionBase : AdvancedSessionExtentionBase
     {
-        protected Dictionary<string, long> TrackedKeys;
-        protected bool Marked;
-
         protected ClusterSessionBase(InMemoryDocumentSessionOperations session) : base(session)
         {
+            if (session.TransactionMode != TransactionMode.ClusterWide)
+            {
+                throw new InvalidOperationException($"This function is part of cluster transaction session, in order to use it you have to open the Session with '{nameof(TransactionMode.ClusterWide)}' option.");
+            }
         }
 
-        protected void CreateCompareExchangeValueInternal<T>(string key, T item)
+        public void CreateCompareExchangeValue<T>(string key, T item)
         {
-            EnsureClusterSession();
-
             if (Session.StoreCompareExchange == null)
                 Session.StoreCompareExchange = new Dictionary<string, (object Item, long Index)>();
 
             // new entity
             Session.StoreCompareExchange.Add(key, (item, 0));
         }
 
-        protected async Task DeleteCompareExchangeValueAsyncInternal(string key, long? index = null)
+        public void DeleteCompareExchangeValue<T>(CompareExchangeValue<T> item)
         {
-            EnsureClusterSession();
-
             if (Session.DeleteCompareExchange == null)
                 Session.DeleteCompareExchange = new Dictionary<string, long>();
 
-            if (TrackedKeys == null)
-                TrackedKeys = new Dictionary<string, long>();
-
-            if (index != null)
-            {
-                if (Session.DeleteCompareExchange.ContainsKey(key) == false)
-                    Session.DeleteCompareExchange.Add(key, index.Value);
-                return;
-            }
+            if (Session.DeleteCompareExchange.ContainsKey(item.Key) == false)
```

Comment:
Missing check for `StoreCompareExchange`.

Where are you checking that we aren't trying to send conflicting commands?